tools/rule_checker/main.go

```diff
@@ -19,29 +19,83 @@ package main
 import (
 	"flag"
 	"fmt"
-
-	"github.com/golang/glog"
+	"io"
+	"os"
 
 	"github.com/prometheus/prometheus/rules"
 )
 
-var ruleFile = flag.String("rule-file", "", "The path to the rule file to check.")
-
-func main() {
-	flag.Parse()
+var (
+	flagset  = flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	ruleFile = flagset.String("rule-file", "", "The path to the rule file to check. (Deprecated)")
+)
 
-	if *ruleFile == "" {
-		glog.Fatal("Must provide a rule file path")
+// checkFile reads from in or the file if in is nil. Sucessfully read rules
+// are printed to out.
+func checkFile(filename string, in io.Reader, out io.Writer) error {
+	if in == nil {
+		f, err := os.Open(filename)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+		in = f
 	}
 
-	rules, err := rules.LoadRulesFromFile(*ruleFile)
+	rules, err := rules.LoadRulesFromReader(in)
 	if err != nil {
-		glog.Fatalf("Error loading rule file %s: %s", *ruleFile, err)
+		return err
 	}
 
-	fmt.Printf("Successfully loaded %d rules:\n\n", len(rules))
-
+	fmt.Fprintf(os.Stderr, "successfully loaded %d rules from %s:\n", len(rules), filename)
 	for _, rule := range rules {
-		fmt.Println(rule)
+		fmt.Fprint(out, rule.String())
+		fmt.Fprint(out, "\n")
 	}
+	return nil
+}
+
+func usage() {
+	fmt.Fprintf(os.Stderr, "usage: rule_checker [path ...]\n")
+
+	flagset.PrintDefaults()
+	os.Exit(2)
+}
+
+func main() {
+	flagset.Usage = usage
+	flagset.Parse(os.Args[1:])
+
+	if flagset.NArg() == 0 && *ruleFile == "" {
+		if err := checkFile("<stdin>", os.Stdin, os.Stdout); err != nil {
+			fmt.Fprintf(os.Stderr, "error checking standard input: %s\n", err)
+			os.Exit(1)
+		}
+		os.Exit(0)
+	}
+
+	paths := flagset.Args()
+	if *ruleFile != "" {
+		paths = []string{*ruleFile}
+
+		fmt.Fprint(os.Stderr, "warning: usage of -rule-file is deprecated.\n")
+	}
+
+	for _, path := range paths {
+		switch dir, err := os.Stat(path); {
+		case err != nil:
+			fmt.Fprintf(os.Stderr, "error checking path %s: %s\n", path, err)
+			os.Exit(2)
+		case dir.IsDir():
+			fmt.Fprintf(os.Stderr, "error checking path %s: directories not allowed\n", path)
+			os.Exit(2)
+		default:
+			if err := checkFile(path, nil, os.Stdout); err != nil {
+				fmt.Fprintf(os.Stderr, "error checking file %s: %s\n", path, err)
+				os.Exit(1)
```

Comment:
Thanks, done.

On the same note, it would be good if multiple errors in a single file would be reported. It's probably better to care about that in the new parser (fingers crossed) rather than hack this now.